suite/tests/api/ir_arm.c

```diff
@@ -111,28 +114,26 @@ test_pred(void *dc)
                       DR_PRED_EQ);
     ASSERT(instr_get_eflags(inst, DR_QUERY_INCLUDE_COND_SRCS) == EFLAGS_READ_ARITH);
     ASSERT(instr_get_eflags(inst, 0) == (EFLAGS_READ_ARITH & (~EFLAGS_READ_GE)));
-    instr_free(dc, inst);
+    instr_destroy(dc, inst);
     inst = INSTR_CREATE_sel(dc, opnd_create_reg(DR_REG_R0), opnd_create_reg(DR_REG_R1),
                             opnd_create_reg(DR_REG_R1));
     ASSERT(instr_get_eflags(inst, DR_QUERY_INCLUDE_COND_SRCS) == EFLAGS_READ_GE);
     ASSERT(instr_get_eflags(inst, 0) == EFLAGS_READ_GE);
-    instr_free(dc, inst);
+    instr_destroy(dc, inst);
     dr_set_isa_mode(dc, old_mode, NULL);
 }
 
 static void
 test_pcrel(void *dc)
 {
-    byte *pc;
     instr_t *inst;
     inst = INSTR_CREATE_ldr(dc, opnd_create_reg(DR_REG_R0),
                             opnd_create_rel_addr((void *)&buf[128], OPSZ_PTR));
-    pc = instr_encode(dc, inst, buf);
-    /* (gdb) x/2i buf+1
-     *    0x120b9 <buf+1>:     ldr     r0, [pc, #124]  ; (0x12138 <buf+128>)
+    /* On decoding our rel-addr opnd turns into a base-disp:
```

Comment:
Is it worth constructing an instruction with the expected base-disp operand and checking equality of `inst`  with that?

Reply:
Duplicating the code from test_instr_encoding?  Maybe it's worth it to avoid the bool param.  OK, done.

Reply:
Not only to avoid the bool param. But also for a stronger test.